### ISSUE TITLE
REFL031 should not warn here

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/ReflectionAnalyzers#162]
```cs
        internal object M<T>()
        {
            if (typeof(T).IsValueType)
            {
                return typeof(Nullable<>).MakeGenericType(typeof(T));
            }
            return null;
        }
```

**Comment by sungam3r** [comment]
Also this false positive:
```c#
 typeof(IDictionary<,>).MakeGenericType(sometype.GetGenericArguments()); 
```

Error	REFL031	The number of generic arguments provided doesn't equal the arity of the generic type definition. The member has 2 parameters but 1 argument are passed in.

**Comment by JohanLarsson** [comment]
@sungam3r your case should be fixed in https://www.nuget.org/packages/ReflectionAnalyzers/0.1.22-dev

**Comment by sungam3r** [comment]
Ok, thanks.

**Comment by JohanLarsson** [comment]
never hesitate to create new issues btw, nothing is too small.
